core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -932,19 +924,18 @@ class TimeSeriesShard(val dataset: Dataset,
   private def updateIndexWithEndTime(p: TimeSeriesPartition,
                                      partFlushChunks: Iterator[ChunkSet],
                                      timeBucket: Int) = {
-    // Synchronize for safe read-modify-write behavior.
-    activelyIngesting.synchronized {
-      if (partFlushChunks.isEmpty && activelyIngesting.get(p.partID)) {
-        var endTime = p.timestampOfLatestSample
-        if (endTime == -1) endTime = System.currentTimeMillis() // this can happen if no sample after reboot
-        updatePartEndTimeInIndex(p, endTime)
-        timeBucketBitmaps.get(timeBucket).set(p.partID)
-        activelyIngesting.clear(p.partID)
-      } else if (partFlushChunks.nonEmpty && !activelyIngesting.get(p.partID)) {
-        // Partition started re-ingesting.
-        updatePartEndTimeInIndex(p, Long.MaxValue)
-        timeBucketBitmaps.get(timeBucket).set(p.partID)
-        activelyIngesting.set(p.partID)
+    if (partFlushChunks.isEmpty && activelyIngesting.get(p.partID)) {
+      // Below is coded to work concurrently with logic in getOrAddPartitionAndIngest
+      // where we try to activate an idle time series
+      activelyIngesting.withWriteLock {
+        // do not use get/set since activelyIngesting locks are not reentrant
+        if (!activelyIngesting.getWithoutLock(p.partID)) {
```

Comment:
Not not? (no ! operator)

Reply:
Ugh! copy/paste error. Thanks for the catch